pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/AdminResource.java

```diff
@@ -376,6 +377,25 @@ protected void validatePartitionedTopicMetadata(String tenant, String namespace,
         }
     }
 
+    protected void validateTopicExistedAndCheckAllowAutoCreation(String tenant, String namespace,
+                                                                 String encodedTopic, boolean checkAllowAutoCreation) {
+        try {
+            PartitionedTopicMetadata partitionedTopicMetadata =
+                    pulsar().getBrokerService().fetchPartitionedTopicMetadataAsync(topicName).get();
+            if (partitionedTopicMetadata.partitions < 1) {
+                if (!pulsar().getNamespaceService().checkTopicExists(topicName).get()
+                        && checkAllowAutoCreation
```

Comment:
Here is not covered by tests? I noticed the tests only cover the `checkAllowAutoCreation=false`, could you please help add a test for cover `checkAllowAutoCreation=true`?

Reply:
I'll add test case to protect this part of code.

Reply:
Thanks